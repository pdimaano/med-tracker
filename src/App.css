main {
  width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 15vh;
  height: calc(100vh - 56px);
  overflow: scroll;
  color: white;
}

.card {
  border-radius: 0;
  margin-bottom: 15vh;
  box-shadow: 10px 5px 3px #000124;
}

.home-card {
  width: 70%;
  margin: auto;
}