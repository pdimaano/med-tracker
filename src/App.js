import React, { useState, useEffect } from "react";
import "./App.css";
import Home from "./Home";
import { fetchMeds, addMed as addMedApi } from "./Api";
import NavBar from "./NavBar";
import { Route, Routes } from "react-router-dom";
import MedList from "./MedList";
import Med from "./Med";
import slugify from "slugify";
import MedForm from "./MedForm";

/** Medication tracker main application.
 *
 *  Props: none
 *
 *  State:
 *  - meds: list of med data objects - populated via AJAX call
 *  - isLoading: boolean, has data loaded yet?
 *
 */

function App() {
  const [isLoading, setIsLoading] = useState(true);
  const [meds, setMeds] = useState({
    sunday: [],
    monday: [],
    tuesday: [],
    wednesday: [],
    thursday: [],
    friday: [],
    saturday: [],
  });

  /** Load data from backend */

  useEffect(function fetchMedsOnMount() {
    async function getAllMeds() {
      const sundays = await fetchMeds("sunday");
      const mondays = await fetchMeds("monday");
      const tuesdays = await fetchMeds("tuesday");
      const wednesdays = await fetchMeds("wednesday");
      const thursdays = await fetchMeds("thursday");
      const fridays = await fetchMeds("friday");
      const saturdays = await fetchMeds("saturday");
      setMeds({
        sundays,
        mondays,
        tuesdays,
        wednesdays,
        thursdays,
        fridays,
        saturdays,
      });
      setIsLoading(false);
    }

    getAllMeds();
  }, []);

  /** Call API to add med for specific day; update state */

  async function addMed(day, { name, description }) {
    let id = slugify(name, { lower: true });
    let objData = { id, name, description };
    await addMedApi(day, objData);
    setMeds((m) => ({
      ...m,
      [day]: [...m[day], objData],
    }));
  }

  /** Show app frame, navbar and routes */

  let {
    sundays,
    mondays,
    tuesdays,
    wednesdays,
    thursdays,
    fridays,
    saturdays,
  } = meds;

  if (isLoading) {
    return <p>Loading...</p>
  }
}
