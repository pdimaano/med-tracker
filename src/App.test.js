import * as api from "./api";
import React from "react";
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import App from './App';

api.fetchMeds = jest.fn();
api.addMed = jest.fn();

const WELCOME_TEXT = "Daily Medication Tracker";

describe("Med Tracker routes", function () {
  beforeEach(() => {
    api.fetchMeds.mockReturnValue([]);
    api.addMed.mockReturnValue([]);
  });

  it("renders without crashing", async function () {
    const { getByText } = render(
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    );
    await waitFor(() => {
      screen.getByText(WELCOME_TEXT);
    });
  });

  it("renders the home snapshot", async function () {
    const { asFragment, debug } = render(
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    );
    await waitFor(() => screen.getByText(WELCOME_TEXT));
    expect(api.fetchMeds).toHaveBeenCalledTimes(7);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the Sunday medications", async function () {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/sunday"]}>
        <App />
      </MemoryRouter>
    );
    await waitFor(() => screen.getByText("Sunday Meds"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the Wednesday medications", async function () {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/wednesday"]}>
        <App />
      </MemoryRouter>
    );
    await waitFor(() => screen.getByText("Wednesday Meds"));
    expect(asFragment()).toMatchSnapshot();
  });
});