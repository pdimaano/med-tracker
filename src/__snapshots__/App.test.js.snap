// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Med Tracker routes renders a medication 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="Navigation navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Track Your Meds
        </a>
        <ul
          class="ms-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/sunday"
            >
              Sunday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="active"
              href="/monday"
            >
              Monday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/tuesday"
            >
              Tuesday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/wednesday"
            >
              Wednesday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/thursday"
            >
              Thursday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/friday"
            >
              Friday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/saturday"
            >
              Saturday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="me-0"
              href="/add"
            >
              Add Med
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <main>
      <section>
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold text-center card-title"
            >
              Test Med
            </div>
            <p
              class="font-italic card-text"
            >
              Test description
            </p>
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Med Tracker routes renders the Sunday medications 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="Navigation navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Track Your Meds
        </a>
        <ul
          class="ms-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="active"
              href="/sunday"
            >
              Sunday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/monday"
            >
              Monday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/tuesday"
            >
              Tuesday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/wednesday"
            >
              Wednesday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/thursday"
            >
              Thursday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/friday"
            >
              Friday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/saturday"
            >
              Saturday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="me-0"
              href="/add"
            >
              Add Med
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <main>
      <section
        class="col-md-4"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold text-center card-title"
            >
              Sunday Meds
            </div>
            <ul
              class="list-group"
            />
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Med Tracker routes renders the Wednesday medications 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="Navigation navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Track Your Meds
        </a>
        <ul
          class="ms-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/sunday"
            >
              Sunday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/monday"
            >
              Monday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/tuesday"
            >
              Tuesday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="active"
              href="/wednesday"
            >
              Wednesday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/thursday"
            >
              Thursday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/friday"
            >
              Friday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/saturday"
            >
              Saturday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="me-0"
              href="/add"
            >
              Add Med
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <main>
      <section
        class="col-md-4"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="fw-bold text-center card-title"
            >
              Wednesday Meds
            </div>
            <ul
              class="list-group"
            />
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Med Tracker routes renders the home snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="Navigation navbar navbar-expand-md"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Track Your Meds
        </a>
        <ul
          class="ms-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/sunday"
            >
              Sunday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/monday"
            >
              Monday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/tuesday"
            >
              Tuesday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/wednesday"
            >
              Wednesday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/thursday"
            >
              Thursday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/friday"
            >
              Friday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/saturday"
            >
              Saturday
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="me-0"
              href="/add"
            >
              Add Med
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <main>
      <section
        class="col-md-8"
      >
        <div
          class="home-card card"
        >
          <div
            class="text-center card-body"
          >
            <div
              class="card-title"
            >
              <p
                class="fw-bold"
                style="font-size: 1.5em;"
              >
                Daily Medication Tracker
              </p>
            </div>
            <p
              class="card-text"
            >
              Never forget your daily meds again!
            </p>
            <img
              alt="Rx symbol"
              class="rx-image"
              src="Rx_symbol.png"
            />
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
